### PR TITLE
Use ko's strict mode for better errors messages

### PR DESCRIPTION
**Pull request by nachocano**
## Proposed Changes

- 🧽 Adopt ko's strict mode for a better on-boarding experience. Based on eventing https://github.com/knative/eventing/pull/2717
-  Updated a yaml file name from `config-br-default-channel.yaml` to `br-default-channel.yaml`

**Release Note**

```release-note
NONE
```

### CODE REVIEW

File: config/400-config-br-default-channel.yaml [hunk 1]
```diff
@@ -1 +1 @@
-core/configmaps/config-br-default-channel.yaml
\ No newline at end of file
+core/configmaps/br-default-channel.yaml
```
Comment (ian-mi) [inline config/400-config-br-default-channel.yaml]: Is this change unrelated?

**Comment by nachocano** [comment]
Yes, miss to update the description. Just trying to make the names
consistent with the others. And probably this messed up the e2e tests

File: test/upload-test-images.sh [hunk 2]
```diff
@@ -27,7 +27,7 @@ function upload_test_images() {
 
   # ko resolve is being used for the side-effect of publishing images,
   # so the resulting yaml produced is ignored.
-  ko resolve ${tag_option} -RBf "${image_dir}" > /dev/null
+  ko resolve --strict ${tag_option} -RBf "${image_dir}" > /dev/null
```
Comment (chizhg) [inline test/upload-test-images.sh]: The path for the test images in eventing was not updated in that PR. So this won't work..

Comment (nachocano) [reply test/upload-test-images.sh]: removed, let's see now

File: config/400-config-br-default-channel.yaml (hunk 1 above)
Comment (nachocano) [reply config/400-config-br-default-channel.yaml]: Done, updated the PR description. Thanks!

**Comment by nachocano** [comment]
/hold
@mattmoor just added the testing images in https://github.com/knative/eventing/pull/2726...

**Comment by chizhg** [comment]
These images are vendored.. You need to update the vendor...

**Comment by nachocano** [comment]
> These images are vendored.. You need to update the vendor for eventing...

Done, let's see now

**Pull request merged**